lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -137,45 +138,10 @@ def self.call_automate(obj, attrs, instance_name, options = {})
       MiqAeEngine.deliver_queue(args, q_options)
     end
   end
+  private_class_method :call_automate
 
-  def self.stream_target(object)
-    case object
-    when EmsEvent
-      object.vm_or_template || object.ext_management_system
-    when VmOrTemplate, MiqServer, Service, ExtManagementSystem, MiqRequest
-      object
-    when Storage
-      object.ext_management_systems.first
-    else
-      object.ext_management_system
-    end
+  def self.tenant_identity_target(obj)
+    obj.respond_to?(:tenant_identity) ? obj : obj.ext_management_system
```

Comment:
What objects don't respond to `tenant_identity`?

would it be best to just add a delegate to those objects
```ruby
delegate :tenant_identity, :to => :ext_management
```